api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -340,6 +340,17 @@ default boolean dropTable(TableIdentifier identifier) {
    */
   Table loadTable(TableIdentifier identifier);
 
+  /**
+   * Register a table.
```

Comment:
Seems from the test, the table is not there and it will create one from the file.  It wasn't too obvious from the name of the API, can we enhance the javadoc a bit to detail that?

Reply:
What happens if the table is already there? Do we throw an exception?

Reply:
Updated the java doc to mention this API will register a table with the catalog if it does not exist. It throws an exception if it exists. I have also added a unit test case.